controller/action/ReportActions.class.php

```diff
@@ -2,8 +2,9 @@
 
 class ReportActions extends Actions
 {
-    public static function executeDmca()
-    {
+
+    /* TODO: Investigate optional route parameters, maybe nullable type hints from php 7.1 help with that. */
+    public static function executeDmca() {
```

Comment:
There are substantial [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) violations between `executeDmca` and `executeDmcaWithClaimId`. Either:

- Unify these so that they are the same action and no code is repeated (preferable if feasible)
- Abstract repeated code into a shared function

Reply:
Since the router doesn't support optional routing and this application uses an EOL (End of Life) PHP version the only thing I was able to do was extracting duplicated code into private methods.

You guys should look into a framework like Symfony for your application so you're more flexible when it comes to such stuff. Also, please update your EOL PHP version! [7.0 is EOL since January 01, 2019.](https://www.php.net/supported-versions.php)